tasks/application_generator.rb

```diff
@@ -0,0 +1,85 @@
+module ActiveAdmin
+  class ApplicationGenerator
+    attr_reader :rails_env, :template, :parallel
+
+    def initialize(opts = {})
+      @rails_env = opts[:rails_env] || 'test'
+      @template = opts[:template] || 'rails_template'
+      @parallel = opts[:parallel]
+    end
+
+    def generate
+      unless correctly_configured_app?
+        puts "App is not correctly configured for running tests #{running_mode}. (Re)building #{app_dir} App. Please wait."
+        system("rm -Rf #{app_dir}")
+      end
+
+      if app_exists?
+        puts "test app #{app_dir} already exists and correctly configured; skipping test app generation"
+      else
+        system "mkdir -p #{base_dir}"
+        args = %W(
+          -m spec/support/#{template}.rb
+          --skip-bootsnap
+          --skip-bundle
+          --skip-gemfile
+          --skip-listen
+          --skip-turbolinks
+          --skip-test-unit
+          --skip-coffee
+        )
+
+        command = ['bundle', 'exec', 'rails', 'new', app_dir, *args].join(' ')
+
+        env = base_env
+        env['INSTALL_PARALLEL'] = 'yes' if parallel
+
+        Bundler.with_original_env { Kernel.system(env, command) }
+
+        rails_app_rake "parallel:load_schema" if parallel
+      end
+    end
+
+    private
+
+    def base_env
+      { 'BUNDLE_GEMFILE' => ENV['BUNDLE_GEMFILE'] }
+    end
+
+    def running_mode
+      parallel ? "in parallel" : "sequentially"
+    end
+
+    def correctly_configured_app?
+      app_exists? && !(parallel ^ parallel_tests_setup?)
```

Comment:
Yeah, absolutely, it's not very straighforward, I'll clarify it!

Something I'm considering though is that we might not need these checks at all? We could _always_ configure the app in parallel (with multiple DBs), and I think sequential tasks would just work too, they would run on the first db.

Reply:
I'll have a look tomorrow at this!

Reply:
Sounds good to me, thanks!

Reply:
Ok, I simplified this in https://github.com/activeadmin/activeadmin/pull/5530/commits/22fdce1ff4e76b381006ca28f1141244e1f0dee1, but now I'm thinking that maybe we don't need the sequential test tasks at all? There's just wrappers on top of rspec / cucumber. If someone wants to run specific specs (or all) sequentially, she can use rspec or cucumber directly?

Reply:
So removed all the sequential tasks and went "parallel by default". One can still run sequential tasks by using `rspec` or `cucumber` directly, but the test tasks use parallelization by default.

Reply:
Ok that's good to know because after reviewing the changes I was going to ask. I think that's a good change. It a big test suite so to get it running even faster is a huge help! Also nice that it's consistent with what and how we run it in CI. 👍🏻

Reply:
So to confirm that I understand right, the rspec and cucumber tests are run in parallel locally against one database?